Classes/DataProcessing/LanguageMenuProcessor.php

```diff
@@ -0,0 +1,333 @@
+<?php
+
+/*
+ * This file is part of the package bk2k/bootstrap-package.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace BK2K\BootstrapPackage\DataProcessing;
+
+use BK2K\BootstrapPackage\Utility\LanguageUtility;
+use TYPO3\CMS\Frontend\DataProcessing\MenuProcessor;
+
+/**
+ * This menu processor utilizes HMENU to generate a json encoded menu
+ * string that will be decoded again and assigned to FLUIDTEMPLATE as
+ * variable. Additional DataProcessing is supported and will be applied
+ * to each record.
+ *
+ * Options:
+ * as - The variable to be used within the result
+ * levels - Number of levels of the menu
+ * expandAll = If false, submenus will only render if the parent page is active
+ * includeSpacer = If true, pagetype spacer will be included in the menu
+ * titleField = Field that should be used for the title
+ *
+ * See HMENU docs for more options.
+ * https://docs.typo3.org/typo3cms/TyposcriptReference/ContentObjects/Hmenu/Index.html
+ *
+ *
+ * Example TypoScript configuration:
+ *
+ * 10 = BK2K\BootstrapPackage\DataProcessing\LanguageMenuProcessor
+ * 10 {
+ *   as = languagenavigation
+ *   expandAll = 1
+ *   includeSpacer = 1
+ *   dataProcessing {
+ *     10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+ *     10 {
+ *        references.fieldName = media
+ *     }
+ *   }
+ * }
+ */
+class LanguageMenuProcessor extends MenuProcessor
+{
+    /**
+     * Gets the language data for the languageUid
+     *
+     * @param int $languageUid
+     * @return string JSON encoded data
+     */
+    protected function getLanguageData($languageUid)
+    {
+        return LanguageUtility::getLanguageData($languageUid);
+    }
+
+    /**
+     * Executes stdWrap on global properties
+     *
+     * @param array $conf
+     */
+    protected function stdWrap(&$conf)
+    {
+        if (isset($conf['languageUid.'])) {
+            $conf['languageUid'] = $this->cObj->stdWrap($conf['languageUid'], $conf['languageUid.']);
+        }
+    }
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->menuDefaults['as'] = 'languagemenu';
+        $this->menuDefaults['titleField'] = '';
+    }
+
+    /**
+     * Add loading from sys_languages in case special.value is not set
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function validateConfiguration()
+    {
+        $this->processorConfiguration = array_replace_recursive(
+            [
+                'special' => 'language',
+                'special.' => [
+                    'value.' => [
+                        'ifBlank.' => [
```

Comment:
we do not need to have all configuration need to be done through typoscript, we should evaluate if the language values are set on php side, and collect sys_languages also here. we can then just insert the generated values into `special.value`.

Reply:
but that's what this code does!? i load the language ids from sys_language if special.value is not set or do i miss something? the only needed change i see here is to use a user object instead of text to be able to difference between with or without sites extension for v9...

Reply:
ah, now i understood. change is pending ;)